waiter/src/waiter/service_description.clj

```diff
@@ -79,6 +79,7 @@
    (s/optional-key "idle-timeout-mins") (s/both s/Int (s/pred #(<= 1 % (t/in-minutes (t/days 30))) 'between-1-minute-and-30-days))
    (s/optional-key "interstitial-secs") (s/both s/Int (s/pred #(<= 0 % (t/in-seconds (t/minutes 60))) 'at-most-60-minutes))
    (s/optional-key "restart-backoff-factor") schema/positive-number-greater-than-or-equal-to-1
+   (s/optional-key "scheduler") schema/non-empty-string
```

Comment:
What happens when this is set, but the corresponding scheduler does not exist? Similarly, what happens when this is set, but we're not using a composite scheduler?

What happens when we *are* using a composite scheduler, but this field isn't specified? Do we give a default value? How does this affect the `service-id` computation?

Reply:
> What happens when this is set, but the corresponding scheduler does not exist? 

An error is reported during service creation on the scheduler (i.e. `create-service-if-new`).

> Similarly, what happens when this is set, but we're not using a composite scheduler?

I will add a defense against this scenario and repsond to this comment. 

> What happens when we are using a composite scheduler, but this field isn't specified? Do we give a default value? 

Yes, the default is configurable in the composite scheduler configuration.

> How does this affect the service-id computation?

If the `scheduler` is not explicitly provided by the user, it is not included in the `service-id` computation.

Reply:
> If the scheduler is not explicitly provided by the user, it is not included in the service-id computation.

Does that mean you end up with two difference services depending on whether you explicitly set the `scheduler` or use the default?

Reply:
Reverted publishing of ids. Explicitly specifying the `scheduler` or using the default will now map to the same service id if the values of the `scheduler` parameter are the same.

Reply:
Is there a test to make sure we don't break this equivalence?

Reply:
Ok, I'll add one.

Reply:
I just verified that specifying a parameter same as the default value **ends up changing the `service-id`**. Iirc, we chose to go down this path earlier as we didn't want to end up changing a user's `service-id` just because we changed defaults.

I addinged integration tests that verifies the change in service-id even when we specify defaults.